selfservice/strategy/oidc/provider_apple.go

```diff
@@ -101,3 +104,37 @@ func (a *ProviderApple) AuthCodeURLOptions(r ider) []oauth2.AuthCodeOption {
 
 	return options
 }
+
+func (a *ProviderApple) Claims(ctx context.Context, exchange *oauth2.Token, query url.Values) (*Claims, error) {
+	claims, err := a.ProviderGenericOIDC.Claims(ctx, exchange, query)
+	if err != nil {
+		return claims, err
+	}
+
+	// https://developer.apple.com/documentation/sign_in_with_apple/sign_in_with_apple_js/configuring_your_webpage_for_sign_in_with_apple#3331292
+	// First name and last name are passed only once in an additional parameter to the redirect URL.
+	// There's no way to make sure they haven't been tampered with. Blame Apple.
+	var user struct {
+		Name *struct {
+			FirstName *string `json:"firstName"`
+			LastName  *string `json:"lastName"`
+		} `json:"name"`
+	}
+	if err := json.Unmarshal([]byte(query.Get("user")), &user); err == nil {
```

Comment:
Could you please add a test for this?

Reply:
@JiggyDown if you could add a small test for this, we can merge this right away :)

Reply:
Test added. Thank you for your patience!